src/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -687,20 +681,53 @@ private String getMolecularBarcode(final List<FastqRecord>  records) {
         }
     }
 
-    /** Reads multiple FASTQs at the same time, ensuring their read names match along the way */
-    class MultiFastqReader implements Iterator<List<FastqRecord>> {
-        public final List<FastqReader> readers = new ArrayList<>();
+    /** Reads multiple FASTQs one after the other. This is useful for when we have more than one FASTQ for a set of reads. */
+    class SequentialFastqReader implements Iterator<FastqRecord> {
```

Comment:
Don't have to do this now, but it might be nice to pull this out and make it implement FastqReader so it can be a drop in replacement for a regular reader.